### DISCUSSION TITLE
Disable deprecations treated like errors?

### DISCUSSION THREAD

**Discussion by andreamancuso** [emscripten-core/emscripten#22146] [General]
Hi,

I'm trying to compile https://github.com/maplibre/maplibre-native using emscripten. I'm trying to retain the existing CMakeLists.txt, though I have had to disable the /WX flag.

I am trying to figure out how to run `emcmake` but I ideally need help figuring out which flags I need to pass. Has anyone ever tried anything like it?

I ran `emcmake cmake . -B build -G Ninja -DCMAKE_BUILD_TYPE=Release -DMLN_WITH_CORE_ONLY=1 -DMLN_WITH_WERROR=OFF` but then I've noticed that the .lib file generated as part of running `cmake --build build` does not contain WASM or LLVM output. How do I force it to use `em++`?

I also tried a different approach, I added add_subdirectory(${DEPS}/maplibre-native) to my CMakeLists.txt so now I am certain that em++ is being used.

Also:

`target_link_libraries(reactDearImgui PRIVATE imgui::imgui implot::implot nlohmann_json::nlohmann_json $<IF:$<TARGET_EXISTS:libuv::uv_a>,libuv::uv_a,libuv::uv> JPEG::JPEG mbgl-core)`

I am getting errors such as

```
C:/dev/react-imgui/packages/dear-imgui/cpp/deps/maplibre-native/vendor/mapbox-base/deps/variant/include\mapbox/variant.hpp:533:37: error: 'aligned_storage<12, 4>' is deprecated [-Werror,-Wdeprecated-declarations]
```

It seems deprecations are treated like errors?

**Comment by sbc100** [comment]
It looks like `aligned_storage` was deprecated in C++23: https://en.cppreference.com/w/cpp/types/aligned_storage.

You can suppress the warning or you can update mapbox to avoid using it.

**Reply by andreamancuso** [reply]
Sorry, I should have RTFM on this one.
